Jenkinsfile

```
@@ -0,0 +1,62 @@
+pipeline {
+    agent any
+
+    stages {
+        stage('Checkout') {
+            steps {
+                checkout scm
+            }
+        }
+        stage('Dotnet Build') {
+            steps {
+                sh "dotnet build DotnetTemplate.sln"
+            }
+        }
+        stage('Dotnet Tests') {
+            steps {
+                sh "dotnet test DotnetTemplate.Web.Tests"
+            }
+        }
+        stage('Typescript Build') {
+            steps {
+                dir("DotnetTemplate.Web/") {
+                    sh "npm ci"
+                    sh "npm run build"
+                }
+            }
+        }
+        stage('Typescript Tests') {
+            steps {
+                dir ("DotnetTemplate.Web/") {
+                    sh "npm test"
+                }
+            }
+        }
+        stage('Typescript Lint') {
```

Isn't linting a relatively quick step?
Wouldn't it be better to put it earlier?

---

Yes it is and sure I could move it earlier. What do you think would be the advantage of that? 🙂

---

To make the build fail faster; don't bother doing anything time consuming if there's a semi-colon in the wrong place (for example).

---

Fair enough, happy to move it around if you prefer